semgrep/semgrep/core_runner.py

```diff
@@ -225,111 +296,58 @@ def _run_rule(
         """
         outputs: List[PatternMatch] = []  # multiple invocations per language
         errors: List[CoreException] = []
-        equivalences = rule.equivalences
 
         for language, all_patterns_for_language in self._group_patterns_by_language(
             [rule]
         ).items():
-            try:
-                targets = target_manager.get_files(
-                    language, rule.includes, rule.excludes
-                )
-            except _UnknownLanguageError as ex:
-                raise UnknownLanguageError(
-                    short_msg="invalid language",
-                    long_msg=f"unsupported language {language}",
-                    spans=[rule.languages_span.with_context(before=1, after=1)],
-                ) from ex
 
+            targets = self.get_files_for_language(language, rule, target_manager)
             if targets == []:
                 continue
 
-            # semgrep-core doesn't know about OPERATORS.REGEX - this is
-            # strictly a semgrep Python feature. Regex filtering is
-            # performed purely in Python code then compared against
-            # semgrep-core's results for other patterns.
-            patterns_regex, patterns = partition(
-                lambda p: p.expression.operator == OPERATORS.REGEX,
-                all_patterns_for_language,
-            )
-            if patterns_regex:
-                patterns_json = [pattern.to_json() for pattern in patterns_regex]
+            if rule._mode == "taint":
+                pattern_json = rule._raw.copy()
+                del pattern_json["mode"]
```

Comment:
do we also support `metadata`?

Reply:
Within search-mode there is still the pattern-regex option, if that's what you're talking about? We don't have metadata as a separate mode.

Reply:
There's an auxiliary `metadata` field outside of `patterns`, which I assume is what Drew is referring to?

Reply:
Ah, great! I just added a check for this to say it isn't supported